react-components/src/data/search/queryBuilder/Validations.js

```diff
@@ -1,26 +1,85 @@
-import React from "react";
-import styles from "../styles";
+import { getIn, setIn } from "formik";
 
-import { withStyles } from "@material-ui/core/styles";
-
-const nonEmptyField = (value) =>
-    value && value.length > 0 ? undefined : "Empty Value";
 const minValue = (value) =>
     value && value < 0 ? `Must be at least 0` : undefined;
-const error = (props) => {
-    let { touched, error, classes } = props;
-    return (
-        <div>
-            {touched &&
-                (error && <span className={classes.errorField}>{error}</span>)}
-        </div>
-    );
-};
 
-const errorField = withStyles(styles)(error);
+const validate = (values) => {
+    let errors = {};
 
-export default {
-    nonEmptyField,
-    minValue,
-    errorField,
+    errors = validateClause(values, null, errors);
+    return errors;
 };
+
+function validateClause(values, path, errors) {
+    let typeField = path ? `${path}.type` : "type";
+    let typeVal = getIn(values, typeField);
+    let argsField = path ? `${path}.args` : "args";
+    let argsVal = getIn(values, argsField);
+
+    switch (typeVal) {
+        case "all":
+        case "any":
+            if (!argsVal || !argsVal.length) {
+                errors = setIn(errors, typeField, "Empty grouping");
```

Comment:
Ah, I think it should be pretty equivalent to `setFieldValue`.  I just happened to find `setIn` before I found `setFieldValue`, so I'm using one in some places and not others.  Formik works really well, but I'm surprised I had to find out about `getIn` and `setIn` through formik's PRs and issues rather than in their documentation since they're both so handy.